### DISCUSSION TITLE
"Faking" Plugin Installation

### DISCUSSION THREAD

**Discussion by astrelsky** [NationalSecurityAgency/ghidra#3413] [Q&A]
Is there a way I can force Ghidra to believe a plugin has been installed when in headless mode? I'm currently in a situation where I have a plugin which is both providing an `ExtensionPoint` and it's implementations. Unfortuantely it seems that if the plugin is not installed through the ui then the `ExtensionPoint` implementations are never returned by the `ClassSearcher` even though they clearly **must** be in the classpath.

**Comment by dragonmacher** [comment]
> Unfortuantely it seems that if the plugin is not installed through the ui then the ExtensionPoint implementations are never returned by the ClassSearcher even though they clearly must be in the classpath.

Seems like this should be working.  Have you had a chance to debug this?

**Reply by astrelsky** [reply]
> > Unfortuantely it seems that if the plugin is not installed through the ui then the ExtensionPoint implementations are never returned by the ClassSearcher even though they clearly must be in the classpath.
> 
> Seems like this should be working. Have you had a chance to debug this?

I am unable to get this to reproduce locally. I have an open issue in my plugin repo for it [#43](https://github.com/astrelsky/Ghidra-Cpp-Class-Analyzer/issues/43). I can only get it to occur in ci which I can't debug.

I suspect that it is because the `ExtensionPoint.manifest` isn't registered to be found or something. I think I could use `ExtensionUtils.install(ResourceFile)` but I don't know how I could obtain the correct directory. This would be something I would make my class that is responsible for retrieving the `ExtensionPoint` implementations handle. So it would be in the same jar or build directory as the plugin is.

I recall having a problem at one point where I had to explicitly add the `data` folder of my plugin project into the classpath or they wouldn't get loaded. (Not using Eclipse, and never will) I no longer have that change and the issue had magically disappeared without explanation.

The only reason I didn't open an issue for this is because there is almost nothing to go off of in terms of for reproducing and debugging. I also wouldn't exactly consider them not being loaded unintended behavior here.
